docs/setting-up-dnschain-namecoin-powerdns-server_ubuntu.md

```diff
@@ -0,0 +1,191 @@
+
+script
+user=<yourusername>
+home=/home/$user
+cmd=/usr/bin/namecoind
+pidfile=$home/.namecoin/namecoind.pid
+# Don't change anything below here unless you know what you're doing
+[[ -e $pidfile && ! -d "/proc/$(cat $pidfile)" ]] && rm $pidfile
+[[ -e $pidfile && "$(cat /proc/$(cat $pidfile)/cmdline)" != $cmd* ]] && rm $pidfile
+exec start-stop-daemon --start -c $user --chdir $home --pidfile $pidfile --startas $cmd -b --nicelevel 10 -m
+end script
+```
+Then use `namecoind stop` to stop the process. Issue `sudo initctl reload-configuration` then restart using `sudo shutdown -r now`.
+Confirm using `top` that Namecoin starts automatically.
+	
+As mentioned, `namecoind` is going to begin downloading the blockchain. We won't be able to lookup domain names from the blockchain until it has
+made some progress, later when you revisit the Namecoin, you can try:
+```
+$ namecoind getinfo
+$ namecoind name_show d/okturtles
+```
+as well as checking the RPC interface (use the *rpcuser* and *rpcpassword* from namecoin.conf)
+```
+$ curl --user rpcuser:rpcpassword --data-binary '{"jsonrpc":"1.0","id":"curltext","method":"getinfo","params":[]}'  -H 'content-type: text/plain;' http://127.0.0.1:8336
+$ curl -v -D - --user rpcuser:rpcpassword --data-binary '{"jsonrpc":"1.0","id":"curltext","method":"name_show","params":["d/okturtles"]}' -H 'content-type: text/plain;' http://127.0.0.1:8336
+```
+## Install PowerDNS
+
+Install [PowerDNS](https://www.powerdns.com/) into the system using
+```
+$ sudo apt-get install pdns-recursor
+```
+The command to interface with the PowerDNS server is `rec_control`, as in
+```
+$ sudo rec_control ping		# check if server is alive
+```
+Next, tell PowerDNS to send requests for `.bit`, `.eth` and `.p2p` domain names to port 5333. This configuration is specified in __/etc/powerdns/recursor.conf__
+```
+forward-zones=bit.=127.0.0.1:5333,dns.=127.0.0.1:5333,eth.=127.0.0.1:5333,p2p.=127.0.0.1:5333
+export-etc-hosts=off
+allow-from=0.0.0.0/0
+local-address=0.0.0.0
+local-port=53
+```
+Notice in particular our *forward-zones* declaration. Make sure you restart PowerDNS at this point using `sudo service pdns-recursor restart`.  Then,
+confirm that PowerDNS can correctly resolve conventional domain names before we move on.
+```
+dig @127.0.0.1 okturtles.com
+```
+You should get a result similar to this, with an IP address found for okturtles.com.
+
+![](http://i.imgur.com/EhVaMUb.png)
+
+   
+## Install DNSChain
+
+First, update apt-get and install some pre-requisites. Note that while `install npm` installs node.js, `nodejs-legacy` is needed because the binary is now `nodejs` instead of `node` and prerequisites of the dnschain install (hiredis?) will ask for `node`. Do not use `sudo apt-get install node` because this `node` is unrelated to node.js. See this [stackoverflow discussion](http://stackoverflow.com/questions/21168141/can-not-install-packages-using-node-package-manager-in-ubuntu) for details.
```

Comment:
all set

On Sat, Apr 4, 2015 at 9:45 PM, E. Timothy Uy <tim@loqu8.com> wrote:

> I'll add the sentence and the README.
>
> On Sat, Apr 4, 2015 at 9:45 PM, Greg Slepak <notifications@github.com>
> wrote:
>
>> In docs/setting-up-dnschain-namecoin-powerdns-server_ubuntu.md
>> <https://github.com/okTurtles/dnschain/pull/150#discussion_r27774773>:
>>
>> > +local-address=0.0.0.0
>> > +local-port=53
>> > +```
>> > +Notice in particular our *forward-zones* declaration. Make sure you restart PowerDNS at this point using `sudo service pdns-recursor restart`.  Then,
>> > +confirm that PowerDNS can correctly resolve conventional domain names before we move on.
>> > +```
>> > +dig @127.0.0.1 okturtles.com
>> > +```
>> > +You should get a result similar to this, with an IP address found for okturtles.com.
>> > +
>> > +![](http://i.imgur.com/EhVaMUb.png)
>> > +
>> > +
>> > +## Install DNSChain
>> > +
>> > +First, update apt-get and install some pre-requisites. Note that while `install npm` installs node.js, `nodejs-legacy` is needed because the binary is now `nodejs` instead of `node` and prerequisites of the dnschain install (hiredis?) will ask for `node`. Do not use `sudo apt-get install node` because this `node` is unrelated to node.js. See this [stackoverflow discussion](http://stackoverflow.com/questions/21168141/can-not-install-packages-using-node-package-manager-in-ubuntu) for details.
>>
>> Ah OK, didn't know that. Well, in that case would you like me to merge
>> now? Or will you push a commit to add yourself to the README?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/okTurtles/dnschain/pull/150/files#r27774773>.
>>
>
>